### ISSUE TITLE
Make StreamPart a public class

### ISSUE THREAD

**Issue by vickichowder** [alexmojaki/s3-stream-upload#7]
Hello, 

I'd like to test some of  my own code that uses the s3 stream upload. To do that I'll need to mock the StreamTransferManager and the MultipartOutputStream, meaning the StreamPart class as well. 

Is it possible to expose the StreamPart class to a public class so that this can be done?

Thank you!

**Comment by alexmojaki** [comment]
I don't think that you're following sensible testing practice. StreamPart shouldn't appear in either your source or testing code. Consider making a couple of `interface`s for the things you need to mock. The real implementation will be a thin wrapper around the library classes. The tests use a class implementing the same interface but with mock methods. Ideally dependency injection would determine the concrete class used depending on whether you're running tests or the real application. Don't inherit from StreamTransferManager or MultipartOutputStream.
